plugins/indexing/aws/sqs_client.go

```diff
@@ -0,0 +1,69 @@
+package pluginaws
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/service/sqs"
+
+	types "github.com/sedaprotocol/seda-chain/plugins/indexing/types"
+)
+
+const (
+	MaxAwsRequestLengthBytes = 262_144 // 256KB
+	MaxSQSBatchSize          = 10
+)
+
+func (sc *SqsClient) PublishToQueue(data []*types.Message) error {
+	sc.logger.Trace("publishing to queue", "size", len(data))
+
+	sizedBatchEntries, err := sc.createSizedBatchEntries(data)
+	if err != nil {
+		return err
+	}
+
+	currentRequestSize := 0
+	entries := make([]*sqs.SendMessageBatchRequestEntry, 0, MaxSQSBatchSize)
+	for _, sizedBatchEntry := range sizedBatchEntries {
+		nextSize := currentRequestSize + sizedBatchEntry.size
```

Comment:
It seems this variable is unnecessary - just a minor suggestion though.

Reply:
You mean you would leave out the variable and 'inline' the addition in the check? Or is there a cleaner way to do it?

Reply:
Yeah inline. Or assign `nextSize` to `currentRequestSize` directly at the end, I guess. Minor suggestion.

Reply:
Sorry if I’m being dense but I’m not sure if I understand correctly. I use nextSize to determine whether adding the current message will make the batch go over the size limit or not so the only place I would need it is in the check jut below. My reasoning to extract from the statement was for legibility, but if this is not good practice I can just move it into the check. :)

Reply:
My bad for confusing wording.. I think we can just change line 39 to:
```
currentRequestSize = nextSize
```